### PR TITLE
Spellblade projectiles have a 10% chance to turn people hit by them into a dullahan, and fixes dullahans gibbing when thrown through the recycler

### LINKED ISSUE #32255: Dullahan explodes if thrown into recycler, being unable to reenter brain
Recycler wasn't emagged.

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
tweak: Spellblade projectiles might now turn people hit by them into dullahans.
/:cl:

Qbopper gave me this idea so I implemented it. I figured it both was a fine way to add them (spellblades are magical and cut off limbs) and implemented a neat feature. It does what it says on the tin, as well as fixing a problem where their perspective wasn't centered on their head until they toggled body-vision on and off.

Fixes #32255

### CODE REVIEW

**Comment by Qbopper** [comment]
nani

**Comment by epicmemename** [comment]
make it like 1%, 10% seems a little too high for a meme

**Comment by praisenarsie** [comment]
fix all the dullahan issues then you are allowed

**Comment by Xhuis** [comment]
I'm fine with changing the values. In testing 10% of projectiles usually critted or killed targets without landing the roll though.

**Comment by praisenarsie** [comment]
https://github.com/tgstation/tgstation/milestone/32

**Comment by Xhuis** [comment]
one's not something that can reasonably be fixed and another is hard to track down, but I'll look into fixing them both

**Comment by Qbopper** [comment]
\>lzimann -1'd the pr

i am actually properly laughing out loud

**Comment by cebutris** [comment]
It should only happen if it hits their head

**Comment by Xhuis** [comment]
not actually sure how to check that, but that's a good idea so I'll try and do that

**Comment by Xhuis** [comment]
I fixed one of the issues (the recycler one) but the other one is an arcane secret that I don't know how to unravel

File: code/game/machinery/recycler.dm [hunk 1]
```diff
@@ -109,6 +109,7 @@
 		var/brain_holder = istype(AM, /obj/item/organ/brain) || (istype(as_head) && as_head.brain) || (istype(as_mmi) && as_mmi.brain) || istype(AM, /mob/living/brain)
 		if(brain_holder)
 			emergency_stop(AM)
+			return
```
[the remark this answers is not in the document]
Comment (Xhuis) [reply code/game/machinery/recycler.dm]: It actually does! I don't know why either, but it does! Putting this return statement here will prevent your head from being destroyed, but not having it there won't.

File: code/modules/mob/living/carbon/human/species_types/dullahan.dm [hunk 2]
```diff
@@ -32,6 +32,7 @@
 		head.throwforce = 25
 		myhead = new /obj/item/dullahan_relay (head, H)
 		H.put_in_hands(head)
+		update_vision_perspective(H)
```
[the remark this answers is not in the document]
Comment (Xhuis) [reply code/modules/mob/living/carbon/human/species_types/dullahan.dm]: Without this, the vision doesn't set itself to the head when the species is assigned, so it just pops your head off but you keep normal vision and hearing with just your body. This forces it to the head perspective.

File: code/game/machinery/recycler.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (KorPhaeron) [reply code/game/machinery/recycler.dm]: I guess it papers over the issue then. The thing people are complaining about is that the head dumps a false brain, which your ID doesn't transfer to on dullahan death. Same issue if someone butchers a dullahans head, their real brain is still in their (about to gib) body

[the remark this answers is not in the document]
Comment (Xhuis) [reply code/game/machinery/recycler.dm]: If I can think of/find a way to fix it more properly, I will. I'm just confused as to why this works to begin with.

**Comment by nicbn** [comment]
Remember, if you monkeyize a dullahan it will die, #32237

File: code/game/machinery/recycler.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicbn) [reply code/game/machinery/recycler.dm]: It works because it is in a loop `line 105 for(var/i in to_eat)` , so it stops the loop.

**Comment by Xhuis** [comment]
If I take your human status off, will you die?

**Comment by optimumtact** [comment]
These are way to buggy to be an obtainable race I think

**Comment by Bawhoppen** [comment]
Then again spellblades do only show up in wizard rounds, so nothing much lost

**Comment by Xhuis** [comment]
@optimumtact Do you think it would be unlikely for this PR to be merged due to the current state of dullahans?

**Comment by Kromgar** [comment]
@Bawhoppen bubblegum drops spellblades

**Comment by Bawhoppen** [comment]
Oh okay

**Comment by optimumtact** [comment]
dullhans are too buggy I think

**Comment by Xhuis** [comment]
darn

**Comment by Kromgar** [comment]
@Xhuis Gotta fix all the dullahan bugs Xhuis and the dream can be real

**Pull request closed without merging**